File: lib/domain/cultivation_plan/dtos/plan_copy_field_cultivation_row.rb
# frozen_string_literal: true

module Domain
  module CultivationPlan
    module Dtos
      class PlanCopyFieldCultivationRow
        attr_reader :cultivation_plan_field_id, :cultivation_plan_crop_id, :area, :status

        def initialize(cultivation_plan_field_id:, cultivation_plan_crop_id:, area:, status:)
          @cultivation_plan_field_id = cultivation_plan_field_id
          @cultivation_plan_crop_id = cultivation_plan_crop_id
          @area = area
          @status = status
        end
      end
    end
  end
end

File: crates/agrr-domain/src/cultivation_plan/dtos/plan_copy_field_cultivation_row.rs
//! Ruby: plan copy field cultivation row.

#[derive(Debug, Clone, PartialEq)]
pub struct PlanCopyFieldCultivationRow {
    pub id: i64,
    pub cultivation_plan_field_id: i64,
    pub cultivation_plan_crop_id: i64,
    pub area: f64,
    pub status: String,
}
